core/src/main/scala/pathy/Path.scala

```diff
@@ -297,13 +297,12 @@ object Path {
 
```

Comment:
On a side note, I think the use of an index here is horrible. It would be much better if the code handling the case for a file passed `segs.init` to the `folder` function which would always return a `Dir`. The code for file can then construct the `FileIn` itself with the directory obtained from `folder`.